solr/core/src/java/org/apache/solr/search/CollapsingQParserPlugin.java

```diff
@@ -279,7 +299,12 @@ public int getCost() {
     }
 
     public String toString(String s) {
-      return s;
+      return "{!collapse field=" + this.collapseField +
+          ", nullPolicy=" + this.nullPolicy.getName() + ", " +
+          this.groupHeadSelector.toString() +
+          (hint == null ? "": ", hint=" + this.hint) +
+          ", size=" + this.size
+          + "}";
```

Comment:
`{!collapse` specifies the query parser, I think we shouldn't include it in toString()
```suggestion
      return "CollapsingPostFilter[field=" + this.collapseField +
          ", nullPolicy=" + this.nullPolicy.getName() + ", " +
          this.groupHeadSelector +
          (hint == null ? "": ", hint=" + this.hint) +
          ", size=" + this.size
          + "]";
```

Reply:
The above suggestion might lead cause this in debug response
```
CollapsingPostFilter([CollapsingPostFilter field=id, nullPolicy=expand, GroupHeadSelector[selectorText=_version_ asc, type=SORT], hint=top_fc, size=5000])

```
This is caused due to how Query is converted to String by `QueryParsing` Class but I think that should be handled separately as issue exists for different query type
 https://github.com/apache/lucene-solr/blob/ab3f1f0d1d0fcdbe58958197b11ba3837795a24a/solr/core/src/java/org/apache/solr/search/QueryParsing.java#L336